### PR TITLE
Resolve missing valuesets from VSAC API

### PR DESCRIPTION
**Pull request by okeefm**
# Summary
This PR adds the ability for fqm-execution to make HTTP requests to a FHIR API to retrieve expanded valuesets that aren't included in the measure bundle. The bulk of the new resolution code is in `src/helpers/ValueSetResolver.ts`.

## New behavior
* Added `-a`/`--vs-api-key` flag to CLI, and `options.vsAPIKey` to library option object
* Added a `ValueSetResolver` to reach out to the valuesets' URLs and retrieve them (or throw an error)
* Valueset parsing now handles expanded valuesets as well as composed ones. Expansions are considered the default if both are present.

## Code changes
* Turned a bunch of functions into async functions
* Added `ValueSetResolver` class to provide valueset resolution via http, as well as checking whether all missing valuesets have been retrieved
* Moved `ValueSetResolver` and `ValueSetHelper` into a `helpers` directory
* Added `ValueSet` expansion conversion to `valueSetsForCodeService` in `ValueSetHelper`
* Moved `getMissingDependentValuesets` into `ValueSetHelper`

# Testing guidance
* run an `npm install` to get new packages
* Run the fqm-execution CLI using your preferred method, with the `test/fixtures/EXM130-7.3.000-bundle-nocodes-missingVS.json` measure bundle. Result should be an error like `Missing the following valuesets: http://cts.nlm.nih.gov/fhir/ValueSet/2.16.840.1.113883.3.464.1003.101.12.1016, and no API key was provided to resolve them`
* Get your NLM API key (instructions in the README), and provide it to the CLI using `-a <api_key>` and the above measure bundle. Result should be a successful calculation.

### CODE REVIEW

**Comment by mgramigna** [comment]
## jest coverage report 🧪

### Total coverage

| Status | Category   | Percentage         | Covered / Total |
| :----: | :--------- | :----------------- | :-------------: |
|   🟡   | Statements | 78.11%             |    1049/1343    |
|   🟡   | Branches   | 61.06%             |     936/1533    |
|   🟡   | Functions  | 79.18% (+1.54% 🔼) |     194/245     |
|   🟡   | Lines      | 78.29%             |     988/1262    |

> Status of coverage: 🟢 - ok, 🟡 - slightly more than threshold, 🔴 - under the threshold


<details><summary>Show new covered files 🌑</summary>


### Coverage of new files

| Status | Filename            | Statements | Branches | Functions | Lines   |
| :----: | :------------------ | :--------- | :------- | :-------- | :------ |
|   🟢   | src/helpers         | 95.4%      | 81.33%   | 92%       | 95.24%  |
|   🟢   | ValueSetResolver.ts | 100%       | 88.89%   | 100%      | 100%    |

> Status of coverage: 🟢 - ok, 🟡 - slightly more than threshold, 🔴 - under the threshold


</details>


<details><summary>Show files with reduced coverage 🔻</summary>


### Reduced coverage

| Status | Filename             | Statements         | Branches          | Functions           | Lines             |
| :----: | :------------------- | :----------------- | :---------------- | :------------------ | :---------------- |
|   🟡   | src                  | 76.01%             | 59.63%            | 77%                 | 76.13%            |
|   🔴   | CalculatorHelpers.ts | 55.56% (-4.86% 🔻) | 46.2% (-4.96% 🔻) | 41.94% (-10.44% 🔻) | 56.69% (-4.6% 🔻) |

> Status of coverage: 🟢 - ok, 🟡 - slightly more than threshold, 🔴 - under the threshold


</details>

File: src/Execution.ts [hunk 1]
```diff
@@ -24,12 +25,34 @@ export function execute(
...
+          ', '
+        )}, and no API key was provided to resolve them`
+      };
+    }
+    const vsr = new ValueSetResolver(options.vsAPIKey || '');
+    const [expansions, errorMessages] = await vsr.getExpansionForValuesetUrls(missingVS);
+
+    if (errorMessages.length > 0) {
+      return {
+        errorMessage: errorMessages.join()
```
Comment (mgramigna) [inline src/Execution.ts:43]: Any way you can join this with a newline? Hard to read when there are multiple back to back
```suggestion
        errorMessage: errorMessages.join('\n')
```

**Review by mgramigna — CHANGES_REQUESTED** [review]

**Comment by mgramigna** [comment]
I removed all valuesets from my EXM130 bundle, and got some of these error messages:

image

This seems like a pretty short timeout.. and I'm assuming you don't set this anywhere. I wonder if we can up this somehow to be able to not fail when needing to expand a bunch of valuesets?

File: package.json [hunk 2]
```diff
@@ -12,7 +12,9 @@
   ],
   "dependencies": {
     "@ahryman40k/ts-fhir-types": "^4.0.32",
+    "@types/axios": "^0.14.0",
```
[the remark this answers is not in the document]
Comment (okeefm) [reply package.json:15]: Good call. Deleted this and it was still fine.

File: README.md [hunk 3]
```diff
@@ -95,6 +97,12 @@ E.g.
 fqm-execution -o reports -m /path/to/measure/bundle.json -p /path/to/patient1/bundle.json > reports.json
 ```
 
+### ValueSets
+
+If the Measure bundle provided doesn't contain all the required `ValueSet` resources (with expansions or composes) to calculate the measure, an API key can be provided to resolve the valuesets from their provided URLs. Currently only tested with valuesets from [The NLM FHIR Valueset API](https://cts.nlm.nih.gov/fhir).
+
+To find your VSAC API key, sign into [the UTS homepage](https://uts.nlm.nih.gov/uts/), click on `My Profile` in the top right, and copy the `API KEY` value from the `UMLS Licensee Profile`.
```
[the remark this answers is not in the document]
Comment (okeefm) [reply README.md:104]: Technically the page is called `UMLS Licensee Profile`, as it's the page for someone who is licensed to use the UMLS

File: src/Execution.ts (hunk 1 above)
Comment (okeefm) [reply src/Execution.ts:43]: Yeah, good call.

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (mgramigna) [reply README.md:104]: lmao pog

[the remark this answers is not in the document]
Comment (mgramigna) [reply README.md:104]: didn't even bother to check

**Review by mgramigna — APPROVED** [review]

**Pull request merged**